server/text_generation_server/models/custom_modeling/flash_llama_modeling.py

```diff
@@ -152,7 +163,13 @@ def forward(
         self.rotary_emb(query, torch.select(kv, dim=1, index=0), cos, sin)
 
         paged_attention.reshape_and_cache(
-            kv[:, 0], kv[:, 1], kv_cache[0], kv_cache[1], slots
+            kv[:, 0],
+            kv[:, 1],
+            kv_cache[0],
+            kv_cache[1],
+            slots,
+            self.kv_cache_dtype,
```

Comment:
This one isn't necessary here, it should already be inferrable from `kv_cache[0]`.